Fix issue #10: Keyword Validation for URL Shortening

khangklj opened the pull request:
1. Add new ignore folders in .gitignore.
2. Add a Javascript code to validate keyword when shortening the URL.
&emsp;If any keyword is not in **Unreserved Characters** [1], then display an error message at a `<p>` element which has `id="keywordError"`, and throw new Error;
&emsp; Otherwise clear the error message (if exists) and proceed next.

_References:_
[1] [Section 2.3](https://www.rfc-editor.org/rfc/rfc3986#section-2.3) in RFC 3986

Some screenshots:
![Example](https://github.com/user-attachments/assets/54d5cd50-67b9-43ea-aca2-34829f9a8608)

Deepak-Perla commented:
Hey, @khangklj ! Thanks for your contribution. It would be great to add an enhancement to specify the maximum character limit. Also, please remember to pull the code regularly to stay updated with the latest changes. Happy hacking!

khangklj commented:
Hi @Deepak-Perla! I have implemented the enhancement you've said before. Here is the changes:

- Add the `maxlength` attribute for the input `id="keyword"`.
- Add the `<span id="keywordLengthIndicator">` (with some CSS styles). It can dynamically change the current length number while the user is entering the keywords.

Updated screenshot:
![Example2](https://github.com/user-attachments/assets/da70d880-0708-4494-b9ab-023309ce7f7d)
